jcl/src/jdk.incubator.foreign/share/classes/jdk/internal/foreign/abi/LayoutStrPreprocessor.java

```diff
@@ -125,7 +125,13 @@ private static StringBuilder preprocessLayout(MemoryLayout targetLayout, boolean
 			SequenceLayout arrayLayout = (SequenceLayout)targetLayout;
 			MemoryLayout elementLayout = arrayLayout.elementLayout();
 			long elementCount = arrayLayout.elementCount().getAsLong();
-			targetLayoutString.append(elementCount).append(':').append(preprocessLayout(elementLayout, isDownCall));
+
+			/* The padding bytes is required in native to parse the native signature during the upcall */
```

Comment:
do we omit padding bytes for downcalls because libffi handles that for us?

Reply:
Yes. that's how it works for downcall.

Reply:
Updated the comment as suggested above.